Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp

```diff
@@ -73,9 +73,9 @@
   #define CPU_ST7920_DELAY_2 DELAY_NS(0)
   #define CPU_ST7920_DELAY_3 DELAY_NS(189)
 #elif defined(ARDUINO_ARCH_STM32)
-  #define CPU_ST7920_DELAY_1 DELAY_NS(0)
-  #define CPU_ST7920_DELAY_2 DELAY_NS(0)
-  #define CPU_ST7920_DELAY_3 DELAY_NS(0)
+  #define CPU_ST7920_DELAY_1 DELAY_NS(300)
+  #define CPU_ST7920_DELAY_2 DELAY_NS(40)
+  #define CPU_ST7920_DELAY_3 DELAY_NS(340)
```

Comment:
These are surprisingly large delays. Are these the smallest that will work?

Reply:
That could be right and isn't surprising at all.
At the 16MHz AVRs one CPU cycle is ~62.5ns. I distributed the instructions to be part of the delay until i minimized the delays. (per ST7920 datasheet half a cycle in DELAY_3).
Here and now shifting, anding and pin setting contributes very little to the delays.